app/src/ui/main-process-proxy.ts

```diff
@@ -108,42 +151,39 @@ export async function showFolderContents(path: string) {
   }
 
   if (!stats.isDirectory()) {
-    log.error(`Trying to get the folder contents of a non-folder at '${path}'`)
+    log.error(`Cannot open non-folder as folder: '${path}'`)
+    // Fall back to showing the item in its parent folder
     await _showItemInFolder(path)
     return
   }
 
-  // On Windows and Linux we can count on a directory being just a
-  // directory.
-  if (!__DARWIN__) {
-    UNSAFE_openDirectory(path)
-    return
+  // On macOS, check if the directory is an app bundle to prevent accidentally
+  // launching applications when we just want to show the folder contents
+  if (__DARWIN__) {
+    const isBundle = await isApplicationBundle(path).catch(err => {
+      log.error(`Failed to check if path is app bundle '${path}'`, err)
+      // Assume it's a bundle out of caution
+      return true
+    })
+
+    if (isBundle) {
+      log.info(
+        `Preventing direct open of '${path}' as it appears to be an application bundle`
+      )
+      // Show the app bundle in its parent folder instead
+      await _showItemInFolder(path)
+      return
+    }
   }
 
-  // On macOS a directory might also be an app bundle and if it is
-  // and we attempt to open it we're gonna execute that app which
-  // it far from ideal so we'll look up the metadata for the path
-  // and attempt to determine whether it's an app bundle or not.
-  //
-  // If we fail loading the metadata we'll assume it's an app bundle
-  // out of an abundance of caution.
-  const isBundle = await isApplicationBundle(path).catch(err => {
-    log.error(`Failed to load metadata for path '${path}'`, err)
-    return true
-  })
-
-  if (isBundle) {
-    log.info(
-      `Preventing direct open of path '${path}' as it appears to be an application bundle`
-    )
-
-    await _showItemInFolder(path)
-  } else {
-    UNSAFE_openDirectory(path)
```

Comment:
I am concerned we are moving away from `UNSAFE_openDirectory` to remove executables being erroneously opened on Windows.